upload/src/addons/SV/ThreadReplyBanner/XF/Service/Thread/Creator.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+
+namespace SV\ThreadReplyBanner\XF\Service\Thread;
+
+
+class Creator extends XFCP_Creator {
+
+	protected $bannerText = null;
+	protected $bannerActive = null;
+
+	public function setReplyBanner($text, $active) {
+		$this->bannerText = $text;
+		$this->bannerActive = $active;
+	}

```

Comment:
Move the creating the banner entity here, and then extend `_validate()`. Call the banner entity's `preSave()` and add any errors to the return result of the parent function.

This will also need to be done to the Replier class